### ISSUE TITLE
LightInject 3.0.2.5 and RegisterAssembly

### ISSUE THREAD

**Issue by monnster** [seesharper/LightInject#172]
Hi!
I've met a strange behavior of RegisterAssembly & GetAllInstances method. It duplicates instance of each interface implementation. Say, I have 2 projects in solution, and in the first one I define:
```
public interface IFoo { }
public class Foo: IFoo { }
```

In the second I create ServiceContainer and ask it to scan the first assembly, then give me all instances of IFoo:
```
var container = new ServiceContainer();
container.RegisterAssembly(typeof(IFoo).Assembly);
var foos = container.GetAllInstances<IFoo>(); // returns 2 Foos
```

However, if I register Foo explicitly, GetAllInstances works correctly.

Tested in FW 4.5 and 4.5.1

**Comment by petersondrew** [comment]
I'm seeing the exact same behavior. If you have any abstract base classes that also implement IFoo it will return an instance for each one of those as well. So if class Foo inherited from abstract FooBase, you would get three instances in your example.
This happens even when filtering out abstract types in the shouldRegister lambda.

**Comment by monnster** [comment]
The new bahavior introduced in 3.0.2.7 fixes the problem as described in #190. Thanks, Bernard!
